services/provider/server/server.go

```diff
@@ -769,3 +765,43 @@ func (s *OCSProviderServer) getOCSSubscriptionChannel(ctx context.Context) (stri
 	}
 	return subscription.Spec.Channel, nil
 }
+
+func (s *OCSProviderServer) getClusterResourceQuotaSpec(consumerResource *ocsv1alpha1.StorageConsumer) quotav1.ClusterResourceQuotaSpec {
+
+	clusterResourceQuotaSpec := &quotav1.ClusterResourceQuotaSpec{
+		Selector: quotav1.ClusterResourceQuotaSelector{
+			LabelSelector: &metav1.LabelSelector{
+				MatchExpressions: []metav1.LabelSelectorRequirement{
+					{
+						Key:      string(consumerResource.UID),
+						Operator: metav1.LabelSelectorOpDoesNotExist,
+					},
+				},
+			},
+		},
+		Quota: corev1.ResourceQuotaSpec{
+			Hard: corev1.ResourceList{"requests.storage": *resource.NewScaledQuantity(
+				int64(consumerResource.Spec.StorageQuotaInGiB),
+				resource.Giga,
+			)},
+		},
+	}
+
+	return *clusterResourceQuotaSpec
+}
+
+func (s *OCSProviderServer) getDesiredClientConfigMap(ctx context.Context, consumerResource *ocsv1alpha1.StorageConsumer) (map[string]string, error) {
```

Comment:
1. This does not return a config map, and should not return one so the name does not fit the function. 

2. You cannot use a map for this, a map has no greatness on order so hashing it might not result in the same hash even if it has the same keys and values. Please use an array of key/value pairs, and make sure that you have a consistent order for the items in the array. 

3. Nowhere in the code there is a need to read into the content of the return value of this map/array. Any and all code outside of this function only cares about the hash of the important part of the consumer's data. Please restructure your code to reflect this - the function should return the hash value and should be named accordingly

Reply:
even with different orders, the md5 hash returns the same hash I have verified that but I agree using array. I will do the change in updated commit.

Reply:
There are different types of hash algo, the fact the md5 is no susceptible to order changes does not mean that other hash algo aren't and we should not assume that we will keep on using md5 in the future